homeassistant/components/androidtv_remote/remote.py

```diff
@@ -0,0 +1,152 @@
+"""Remote control support for Android TV Remote."""
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Iterable
+import logging
+from typing import Any
+
+from androidtvremote2 import AndroidTVRemote, ConnectionClosed
+
+from homeassistant.components.remote import (
+    ATTR_ACTIVITY,
+    ATTR_DELAY_SECS,
+    ATTR_HOLD_SECS,
+    ATTR_NUM_REPEATS,
+    DEFAULT_DELAY_SECS,
+    DEFAULT_HOLD_SECS,
+    DEFAULT_NUM_REPEATS,
+    RemoteEntity,
+    RemoteEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+PARALLEL_UPDATES = 0
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Android TV remote entity based on a config entry."""
+    api: AndroidTVRemote = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([AndroidTVRemoteEntity(api, config_entry)])
+
+
+class AndroidTVRemoteEntity(RemoteEntity):
+    """Representation of an Android TV Remote."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    def __init__(self, api: AndroidTVRemote, config_entry: ConfigEntry) -> None:
+        """Initialize device."""
+        self._api = api
+        self._host = config_entry.data[CONF_HOST]
+        self._name = config_entry.data[CONF_NAME]
+        self._attr_unique_id = config_entry.unique_id
+        self._attr_supported_features = RemoteEntityFeature.ACTIVITY
+        self._attr_is_on = api.is_on
+        self._attr_current_activity = api.current_app
+        device_info = api.device_info
+        self._attr_device_info = DeviceInfo(
+            connections={(CONNECTION_NETWORK_MAC, config_entry.data[CONF_MAC])},
+            identifiers={(DOMAIN, str(config_entry.unique_id))},
+            name=self._name,
+            manufacturer=None if not device_info else device_info["manufacturer"],
+            model=None if not device_info else device_info["model"],
```

Comment:
Seems to be that if api.device_info is None, it probably shouldn't even setup the entity? Is there a case where device_info in your api is None even after being correctly set up?

Reply:
Added assert